### ISSUE TITLE
Feature Request: user add directives

### ISSUE THREAD

**Issue by nodrygo** [diml/utop#162]
Should be great if a user was able to add some personal directives on the utop config file
I would love to see some directives  like this
\#ls
\#pwd
\#edit <file> (* with vim or emacs choice *) 
\#!  <external os command> 
\#doc 'xxx'
ect.....
Is it possible ? 
thank's

**Comment by diml** [comment]
Yes, you can do it using the [Toploop.add_directive](https://github.com/ocaml/ocaml/blob/trunk/toplevel/toploop.mli#L51) from OCaml. Regarding the external editor, in the development version of lambda-term, you can do `C-x C-e` to start an external editor, so this will be available in the next version

**Comment by nodrygo** [comment]
Thank you
I have yet to understand how to do but the answer is clear

**Comment by nilsbecker** [comment]
this is nice, but some more default directives would be useful i find. i have been missing `#ls`, and as a stop-gap `#!` sorely.
